examples/earth-and-mars/index.js

```diff
@@ -0,0 +1,202 @@
+
+        context.fillStyle = grd;
+    });
+
+const canvasLine = fc
+    .seriesCanvasLine()
+    .mainValue(mainValue)
+    .crossValue(crossValue);
+
+const canvasSeries = fc.seriesCanvasMulti().series([canvasArea, canvasLine]);
+
+const requestRedraw = () => {
+    d3.select('d3fc-group')
+        .node()
+        .requestRedraw();
+};
+
+const zoom = d3.zoom().on('zoom', () => {
+    x.domain(d3.event.transform.rescaleX(x2).domain());
+    requestRedraw();
+});
+
+const decorate = sel => {
+    sel.select('.plot-area')
+        .on('measure.range', () => {
+            x2.range([0, d3.event.detail.width]);
+        })
+        .call(zoom);
+};
+
+const multiColorAreaDecorate = program => {
+    program.vertexShader().appendHeader(`varying lowp vec4 vColor;`)
+        .appendBody(`float colourModifier = smoothstep(50.0, 400.0, aMainValue);
+        vColor = (vec4(0.55, 0.65, 0.75, 1) * colourModifier) + ((1.0 - colourModifier) * vec4(0.75, 0.45, 0.45, 1));
+        float verticalFade = max(0.0, smoothstep(-1.1, -0.9, gl_Position.y) - 0.15);
+        vColor.a = vColor.a * verticalFade;
+    `);
+
+    program.fragmentShader().appendHeader(`
+        varying lowp vec4 vColor;
+    `).appendBody(`
+        gl_FragColor = vColor;
+    `);
+};
+
+const multiColorLineDecorate = program => {
+    program.vertexShader().appendHeader(`varying lowp vec4 vColor;`)
+        .appendBody(`float colourModifier = smoothstep(-0.875, 1.0, gl_Position.y);
+            vColor = (vec4(0.55, 0.65, 0.75, 1) * colourModifier) + ((1.0 - colourModifier) * vec4(0.75, 0.45, 0.45, 1));`);
+
+    program.fragmentShader().appendHeader(`varying lowp vec4 vColor;`)
+        .appendBody(`
+            gl_FragColor = vColor;
+    `);
+};
+
+const addShowMultiColor = () => {
+    gui.add(options, 'showMultiColor').onChange(render);
+};
+
+const render = () => {
+    currentData = data.slice(0, options.numPoints);
+    updateSeries();
+    decorateWebglSeries();
+
+    chart
+        .canvasPlotArea(options.chartType === 'Canvas' ? canvasSeries : null)
+        .webglPlotArea(options.chartType === 'WebGL' ? webglSeries : null);
+
+    d3.select('#chart')
+        .datum(currentData)
+        .call(chart);
+
+    zoom.transform(d3.select('.plot-area'), d3.zoomIdentity);
+};
+
+// eslint-disable-next-line no-undef
+const gui = new dat.GUI();
+gui.add(options, 'numPoints')
+    .min(2)
+    .max(options.numPoints)
+    .onFinishChange(render);
+gui.add(options, 'chartType', ['WebGL', 'Canvas']).onFinishChange(chartType => {
+    switch (chartType) {
+        case 'WebGL':
+            addShowMultiColor();
+            break;
+        case 'Canvas':
+            gui.__controllers[2].remove();
+            options.showMultiColor = false;
+            break;
+        default:
+            throw new Error('Invalid chart type used.');
+    }
+
+    render();
+});
+addShowMultiColor();
+
+const chart = fc
+    .chartCartesian(x, y)
+    .yOrient('left')
+    .yLabel('Distance (m KM)')
```

Comment:
Usually, units of measure are given in Propercase, so this would read `m Km`. However, assuming this means "million kilometres`, this could be written as "giga-metres" `Gm`.

Reply:
Perhaps `million Km` works best.

Reply:
As discussed this will be changed to `million Km` for clarity